gtsam/navigation/NavState.cpp

```diff
@@ -228,18 +229,27 @@ Vector9 NavState::coriolis(double dt, const Vector3& omega, bool secondOrder,
     dP(xi) -= (0.5 * dt2) * omega_cross2_t;
     dV(xi) -= dt * omega_cross2_t;
   }
+
+  // NOTE(tim): position and velocity changes dP and dV are in nav frame; need
+  // to put them into local frame for subsequent retract
+  dP(xi) = bRn * dP(xi);
+  dV(xi) = bRn * dV(xi);
```

Comment:
Not loving this imperative code here now, so I propose you build n_xi above (you forgot one unrotate, BTW), and then here:
```
// Convert to body-coordinates:
Vector9 xi;
xi << bRn * dR(n_xi), bRn * dP(n_xi), bRn * dV(n_xi);
```

Makes it clearer overall as well.

Reply:
I don't think I forgot one unrotate, since dR was already unrotated in your original code. My style was meant to minimize the changes to existing code. But since you asked, I have pushed a commit that I think makes it clearer what is going on, mostly as you suggest.